Zend/zend_string.h

```diff
@@ -338,23 +338,18 @@ static zend_always_inline void zend_string_release_ex(zend_string *s, bool persi
 	}
 }
 
-#if defined(__GNUC__) && (defined(__i386__) || (defined(__x86_64__) && !defined(__ILP32__)))
-BEGIN_EXTERN_C()
-ZEND_API bool ZEND_FASTCALL zend_string_equal_val(zend_string *s1, zend_string *s2);
-END_EXTERN_C()
-#else
-static zend_always_inline bool zend_string_equal_val(zend_string *s1, zend_string *s2)
+static zend_always_inline bool zend_string_equal_content(const zend_string *s1, const zend_string *s2)
 {
+	if (ZSTR_LEN(s1) != ZSTR_LEN(s2)) {
+		return false;
+	}
+	if (ZSTR_H(s1) && ZSTR_H(s2) && ZSTR_H(s1) != ZSTR_H(s2)) {
+		return false;
+	}
```

Comment:
zend_string_equal_content() is used in zend_hash_find() when hash values are already compared. May be this additional checks lead to degradation on ``$x = $hash['v']`` and Wordpress.

Reply:
I hope the latest commit removes the duplicate hash check correctly.

Prior to the change:
```
Run1: $x = $hash['v']          0.224    0.204
Run2: $x = $hash['v']          0.223    0.204
Run3: $x = $hash['v']          0.221    0.202
```

After the change:
```
Run1: $x = $hash['v']          0.201    0.182
Run2: $x = $hash['v']          0.204    0.186
Run3: $x = $hash['v']          0.201    0.182
```